sig-scalability/slos/network_latency.md

```diff
@@ -0,0 +1,47 @@
+## In-cluster network latency SLIs/SLOs details
+
+### Definition
+
+| Status | SLI | SLO |
+| --- | --- | --- |
+| __WIP__ | In-cluster network latency from a single node, measured as latency of per second ping to kubernetes.default.svc.cluster.local/, measured as 99th percentile over last 5 minutes. | In default Kubernetes installataion with RTT between nodes <= Y, 99th percentile of (99th percentile over all nodes) per cluster-day <= X |
+
+### User stories
+- As a user of vanilla Kubernetes, I want some guarantee how fast my http
```

Comment:
We can have this SLI, but it may not be that useful by itself, for example, TCP throughput is a function of loss + RTT.

Also, in your user story, network latency is probably the smallest contribution to the e2e latency (server time/OS time will dominate).

Reply:
Now that you remind me of the chaining that we do with IP tables, the SLI makes sense :-)

Reply:
Oh -- are you measuring DNS specifically? Shouldn't that be under DNS service SLI, rather than generate networking SLI?

Reply:
I was already thinking about this and my rationale was:
- users care about end-to-end flow (and in many cases it contains dns resolution)
- dns sli would be useful for debugging (i.e. deciding whether it is dns that is slow or other parts of stack). But that's more for developers, not something that we need to expose to users in a form of SLO guarantee

WDYT?

Reply:
We discussed offline and decided to split those. I left a TODO to consider joining them longer-term.